server/src/main/java/org/apache/iotdb/db/utils/QueryDataSetUtils.java

```diff
@@ -202,6 +202,10 @@ public static TSQueryDataSet convertTsBlockByFetchSize(
         break;
       }
       TsBlock tsBlock = optionalTsBlock.get();
+      if (tsBlock.isEmpty()) {
+        break;
+      }
```

Comment:
We cannot simply break here, there is no guarantee that tsBlock won't be empty. If it's really finished, optionalTsBlock.isPresent() will return false.

Reply:
fixed